enterprise/server/usage/usage.go

```diff
@@ -273,28 +273,80 @@ func (ut *tracker) flushCounts(ctx context.Context, groupID string, c collection
 	}
 	dbh := ut.env.GetDBHandle()
 	return dbh.TransactionWithOptions(ctx, db.Opts().WithQueryName("upsert_usage"), func(tx *db.DB) error {
+		// First check whether the row already exists. Make sure to select for
+		// update in order to lock the row.
+		rows, err := tx.Raw(`
+			SELECT *
+			FROM "Usages"
+			WHERE
+				region = ?
+				AND group_id = ?
+				AND period_start_usec = ?
+			`+dbh.SelectForUpdateModifier(),
+			pk.Region,
+			pk.GroupID,
+			pk.PeriodStartUsec,
+		).Rows()
+		if err != nil {
+			return status.WrapError(err, "query rows")
+		}
+
+		schema, err := db.TableSchema(tx, &tables.Usage{})
+		if err != nil {
+			return status.WrapError(err, "schema")
+		}
+
+		existingRowCount := 0
+		for rows.Next() {
+			existingRowCount++
+			fields := map[string]any{}
+			if err := tx.ScanRows(rows, &fields); err != nil {
+				return status.WrapError(err, "scan row")
+			}
+
+			unsupportedField := ""
+			var unsupportedFieldValue any
+			for f, v := range fields {
+				if v == nil || v == "" {
+					continue
+				}
+				if _, ok := schema.FieldsByDBName[f]; !ok {
+					unsupportedField = f
+					unsupportedFieldValue = v
+					break
+				}
+			}
+			if unsupportedField != "" {
+				log.Warningf("Existing usage row contains unsupported column %q with nonempty value %q; dropping usage data.", unsupportedField, unsupportedFieldValue)
```

Comment:
could we use alert.UnexpectedEvent here too?

Reply:
alert.UnexpectedEvent("usage_update_dropped",

Reply:
good catch - done